engine/engine.go

```diff
@@ -156,7 +158,11 @@ func Start(ctx context.Context, startOpts *Config, fn StartCallback) error {
 		return err
 	})
 	eg.Go(func() error {
-		warn, err := progressui.DisplaySolveStatus(context.TODO(), "", nil, os.Stderr, ch)
+		if startOpts.ProgressWriter == nil {
+			startOpts.ProgressWriter = io.Discard
```

Comment:
nit: I think it'd be good to avoid mutating the input param.

I usually do this:

```go
w := startOpts.ProgressWriter
if w == nil {
  w = io.Discard
}
```

Reply:
I hesitate with that too. You're right about mutating.